src/Uno.UI/UI/Xaml/Controls/Repeater/FlowLayout.cs

```diff
@@ -0,0 +1,441 @@
+
+		FlowLayoutAnchorInfo IFlowLayoutAlgorithmDelegates.Algorithm_GetAnchorForRealizationRect(
+			Size availableSize,
+			VirtualizingLayoutContext context)
+		{
+			return GetAnchorForRealizationRect(availableSize, context);
+		}
+
+		FlowLayoutAnchorInfo IFlowLayoutAlgorithmDelegates.Algorithm_GetAnchorForTargetElement(
+			int targetIndex,
+
+			Size availableSize,
+			VirtualizingLayoutContext context)
+		{
+			return GetAnchorForTargetElement(targetIndex, availableSize, context);
+		}
+
+		Rect IFlowLayoutAlgorithmDelegates.Algorithm_GetExtent(
+			Size availableSize,
+			VirtualizingLayoutContext context,
+			UIElement firstRealized,
+			int firstRealizedItemIndex,
+			Rect firstRealizedLayoutBounds,
+			UIElement lastRealized,
+			int lastRealizedItemIndex,
+			Rect lastRealizedLayoutBounds)
+		{
+			return GetExtent(
+				availableSize,
+				context,
+				firstRealized,
+				firstRealizedItemIndex,
+				firstRealizedLayoutBounds,
+				lastRealized,
+				lastRealizedItemIndex,
+				lastRealizedLayoutBounds);
+		}
+
+		void IFlowLayoutAlgorithmDelegates.Algorithm_OnElementMeasured(
+			UIElement element,
+			int index,
+			Size availableSize,
+			Size measureSize,
+			Size desiredSize,
+			Size provisionalArrangeSize,
+			VirtualizingLayoutContext context)
+		{
+			OnElementMeasured(
+				element,
+				index,
+				availableSize,
+				measureSize,
+				desiredSize,
+				provisionalArrangeSize,
+				context);
+		}
+
+		void IFlowLayoutAlgorithmDelegates.Algorithm_OnLineArranged(
+			int startIndex,
+			int countInLine,
+			double lineSize,
+			VirtualizingLayoutContext context)
+		{
+			OnLineArranged(
+				startIndex,
+				countInLine,
+				lineSize,
+				context);
+		}
+
+#endregion
+
+		void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			var property = args.Property;
+			if (property == OrientationProperty)
+			{
+				var orientation = (Orientation)args.NewValue;
+
+				//Note: For FlowLayout Vertical Orientation means we have a Horizontal ScrollOrientation. Horizontal Orientation means we have a Vertical ScrollOrientation.
+				//i.e. the properties are the inverse of each other.
+				ScrollOrientation scrollOrientation = (orientation == Orientation.Horizontal) ? ScrollOrientation.Vertical : ScrollOrientation.Horizontal;
+				ScrollOrientation = scrollOrientation;
+			}
+			else if (property == MinColumnSpacingProperty)
+			{
+				m_minColumnSpacing = (double)args.NewValue;
+			}
+			else if (property == MinRowSpacingProperty)
+			{
+				m_minRowSpacing = (double)args.NewValue;
+			}
+			else if (property == LineAlignmentProperty)
+			{
+				m_lineAlignment = (FlowLayoutLineAlignment)args.NewValue;
+			}
+
+			InvalidateLayout();
+		}
+
+#region private helpers
+
+		double GetAverageLineInfo(
+
+			Size availableSize,
+			VirtualizingLayoutContext context,
+			FlowLayoutState flowState,
+			double avgCountInLine)
+		{
+			// default to 1 item per line with 0 size
+			double avgLineSize = 0;
+			avgCountInLine = 1;
```

Comment:
Codacy found an issue: [Introduce a new variable instead of reusing the parameter 'avgCountInLine'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)